src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -458,10 +458,21 @@ internal static string GetSharedListenerIdOrNull(ITriggerBinding triggerBinding)
 
         internal static void CheckRetrySupport(ITriggerBinding triggerBinding, FunctionDescriptor descr, ILogger logger)
         {
-            if (descr.RetryStrategy != null && triggerBinding != null && triggerBinding.GetType().GetCustomAttribute<SupportsRetryAttribute>() == null)
+            if (descr.RetryStrategy != null)
             {
-                logger?.LogWarning($"Retries are not supported by the trigger binding for { descr.ShortName}.");
-                descr.RetryStrategy = null; // Need to fix tests
+                string functionName = descr?.ShortName;
+                string triggerBindingName = triggerBinding?.GetType().Name;
+                string strategy = JsonConvert.SerializeObject(descr?.RetryStrategy);
```

Comment:
Minor: this is assured not to be null, so can simplify to:
```suggestion
                string strategy = JsonConvert.SerializeObject(descr.RetryStrategy);
```